src/report/mod.rs

```diff
@@ -123,43 +144,66 @@ pub(crate) fn write_csv_detail(
     diffs: &[DiffType],
     rule_name: &str,
 ) -> Result<FileCompareResult, Error> {
+    let nominal_file_name = get_file_name(nominal.as_ref())?.to_string();
+    let actual_file_name = get_file_name(actual.as_ref())?.to_string();
+
     let mut result = FileCompareResult {
-        nominal: nominal.as_ref().to_string_lossy().to_string(),
-        actual: actual.as_ref().to_string_lossy().to_string(),
+        nominal: nominal_file_name,
+        actual: actual_file_name,
         is_error: false,
         detail_path: None,
     };
 
-    let headers: Vec<_> = nominal_table
+    let mut headers: CSVReportRow = CSVReportRow {
+        columns: vec![],
+        has_diff: false,
+        has_error: false,
+    };
+
+    nominal_table
         .columns
         .iter()
         .zip(actual_table.columns.iter())
-        .map(|(n, a)| CSVReport {
-            actual_value: a
-                .header
-                .as_deref()
-                .unwrap_or("Header preprocessing not enabled in config")
-                .to_owned(),
-            nominal_value: n
-                .header
-                .as_deref()
-                .unwrap_or("Header preprocessing not enabled in config")
-                .to_owned(),
-            diffs: Vec::new(),
-        })
-        .collect();
+        .for_each(|(n, a)| {
+            let a_header = a.header.as_deref();
+            let n_header = n.header.as_deref();
+
+            if a_header.is_some() && n_header.is_some() {
```

Comment:
I don't get this change - if you only want to do this in case header is_some, you can use `if let` for this, then you can save the `unwrap_or` in line 173 and 177

Reply:
changed to :
`if let (Some(a_header), Some(n_header)) = (a_header, n_header) {`